plugin/src/main/cpp/extensions/openxr_fb_environment_depth_extension_wrapper.cpp

```diff
@@ -0,0 +1,625 @@
+	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "supports_environment_depth", PROPERTY_HINT_NONE, ""), "", "get_supports_environment_depth");
+	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "supports_hand_removal", PROPERTY_HINT_NONE, ""), "", "get_supports_hand_removal");
+	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "started", PROPERTY_HINT_NONE, ""), "set_started", "get_started");
+	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "enable_hand_removal", PROPERTY_HINT_NONE, ""), "set_enable_hand_removal", "get_enable_hand_removal");
+}
+
+void OpenXRFbEnvironmentDepthExtensionWrapper::cleanup() {
+	if (swapchain != nullptr) {
+		xrDestroyEnvironmentDepthSwapchainMETA(swapchain);
+		swapchain = nullptr;
+	}
+
+	for (auto rid : swapchain_image_rids) {
+		RenderingServer::get_singleton()->free_rid(rid);
+	}
+
+	if (provider != nullptr) {
+		xrDestroyEnvironmentDepthProviderMETA(provider);
+		provider = nullptr;
+	}
+
+	fb_environment_depth_ext = false;
+}
+
+uint64_t OpenXRFbEnvironmentDepthExtensionWrapper::_set_system_properties_and_get_next_pointer(void *p_next_pointer) {
+	system_environment_depth_properties.next = p_next_pointer;
+	return reinterpret_cast<uint64_t>(&system_environment_depth_properties);
+}
+
+Dictionary OpenXRFbEnvironmentDepthExtensionWrapper::_get_requested_extensions() {
+	Dictionary result;
+	for (auto ext : request_extensions) {
+		uint64_t value = reinterpret_cast<uint64_t>(ext.value);
+		result[ext.key] = (Variant)value;
+	}
+	return result;
+}
+
+void OpenXRFbEnvironmentDepthExtensionWrapper::_on_instance_created(uint64_t instance) {
+	if (fb_environment_depth_ext) {
+		bool result = initialize_fb_environment_depth_extension((XrInstance)instance);
+		if (!result) {
+			UtilityFunctions::print("Failed to initialize fb_environment_depth extension");
+			fb_environment_depth_ext = false;
+		}
+	}
+}
+
+void OpenXRFbEnvironmentDepthExtensionWrapper::_on_instance_destroyed() {
+	cleanup();
+}
+
+bool OpenXRFbEnvironmentDepthExtensionWrapper::initialize_fb_environment_depth_extension(const XrInstance p_instance) {
+	GDEXTENSION_INIT_XR_FUNC_V(xrCreateEnvironmentDepthProviderMETA);
+	GDEXTENSION_INIT_XR_FUNC_V(xrDestroyEnvironmentDepthProviderMETA);
+	GDEXTENSION_INIT_XR_FUNC_V(xrStartEnvironmentDepthProviderMETA);
+	GDEXTENSION_INIT_XR_FUNC_V(xrStopEnvironmentDepthProviderMETA);
+	GDEXTENSION_INIT_XR_FUNC_V(xrSetEnvironmentDepthHandRemovalMETA);
+	GDEXTENSION_INIT_XR_FUNC_V(xrCreateEnvironmentDepthSwapchainMETA);
+	GDEXTENSION_INIT_XR_FUNC_V(xrGetEnvironmentDepthSwapchainStateMETA);
+	GDEXTENSION_INIT_XR_FUNC_V(xrDestroyEnvironmentDepthSwapchainMETA);
+	GDEXTENSION_INIT_XR_FUNC_V(xrEnumerateEnvironmentDepthSwapchainImagesMETA);
+	GDEXTENSION_INIT_XR_FUNC_V(xrAcquireEnvironmentDepthImageMETA);
+
+	return true;
+}
+
+bool OpenXRFbEnvironmentDepthExtensionWrapper::prepare() {
+	// TODO: Verify that the shader parameters exist, if possible
```

Comment:
It would possibly be nice if the extension could create the shader parameters, but they need to exist when editing the shader. Would at least be nice to check if they exist, but the relevant methods seem to only be intended from use in the editor.